# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1219244367 14400
#      Wed Aug 20 10:59:27 2008 -0400
# Branch release-3-0-x
# Node ID 72ceeae0f6b23589a3f7c44e9e367482b1140395
# Parent  2ca993580acaf32d052b08ef8d9d4baa0dae72cc
Fix for meshed surfaces with more than one oobject per plot

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-08-20  David Bateman  <dbateman@free.fr>
+
+	* plot/__go_draw_axes__.m: Don't set pm3d implicit if the plot
+	contains a surface. Fixes things like meshc(peaks()).
+
 2008-08-19  Jaroslav Hajek <highegg@gmail.com>
 
 	* optimization/glpk.m: Fix invalid call to zeros.
 
 2008-08-19  David Bateman  <dbateman@free.fr>
 
 	statistics/base/ranks.m: Doc fix.
 
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -996,17 +996,17 @@ function __go_draw_axes__ (h, plot_strea
 	  error ("__go_draw_axes__: unknown object class, %s",
 		 obj.type);
       endswitch
 
     endfor
 
     ## This is need to prevent warnings for rotations in 3D plots, while
     ## allowing colorbars with contours..
-    if (nd == 2 || data_idx > 1)
+    if (nd == 2 || (data_idx > 1 && !view_map))
       fputs (plot_stream, "set pm3d implicit;\n");
     else
       fputs (plot_stream, "set pm3d explicit;\n");
     endif
 
     if (isnan(hidden_removal) || hidden_removal)
       fputs (plot_stream, "set hidden3d;\n");
     else
